pkg/config/nodetreemodel/config.go

```diff
@@ -240,19 +262,24 @@ func (c *ntmConfig) AllKeysLowercased() []string {
 }
 
 func (c *ntmConfig) leafAtPath(key string) LeafNode {
-	pathParts := strings.Split(key, ".")
+	pathParts := strings.Split(strings.ToLower(key), ".")
+
 	curr := c.root
-	for _, part := range pathParts {
+	for _, part := range pathParts[0 : len(pathParts)-2] {
 		next, err := curr.GetChild(part)
 		if err != nil {
-			return &missingLeaf
+			return missingLeaf
+		}
+		if inner, ok := next.(InnerNode); ok {
```

Comment:
We're going to have to type assert either way when traversing trees. It seems simpler to me to always have `GetChild` return a `Node` interface, and then cast once to a `LeafNode` at the end, rather than casting to `InnerNode` every step of the tree traversal.

Reply:
`GetChild` already returns a `Node` but the method only part of the `InnerNode` interface so we need the cast check.

We could add the method to the `LeafNode` too but I felt that the smaller the interface for each node the better. 
Adding `GetChild` to the `Node` interface and `LeafNode` implementation would make us loop one more time to still return an error. My reasoning was to keep the interfaces at a minimal and check for error as soon as possible. WDYT ?

Reply:
I guess it's two ways to accomplish the same thing. Either (1) the general `Node` interface has a `GetChild` method and some implementations (like `LeafNode`) use a method that always fails, or (2) we keep `GetChild` on the `InnerNode` interface only and have to cast more often. I agree that having a smaller interface is usually better, but not necessarily in cases where it makes the calling code more complex.

It may not matter much in the end. But my preference is for the (1) approach because `GetChild` already has the possibility to fail (due to a child at the given key not existing) and I see the `LeafNode` as a specialization that just happens to fail for any key. I don't think this block of code (`leafAtPath`) needs to know the details about different node types, and its simpler just to have it loop `GetChild` calls until it lands where it needs to be.